blmath/geometry/primitives/plane.py

```diff
@@ -245,6 +242,11 @@ def mesh_xsection(self, m, neighborhood=None):
             return Polyline(None)
         return Polyline(np.vstack([x.v for x in components]), closed=True)
 
+    def mesh_intersecting_faces(self, m):
```

Comment:
Sometimes when profiling in python it's useful to break a large function up into many small ones, so that the profiler can figure out which part of a large function has the slow code. Mostly I removed the subfunctions afterwards, because they didn't add clarity, but some of them turned out to be improvements and got left in.